src/app/clusters/window-covering-server/window-covering-server.cpp

```diff
@@ -697,7 +643,26 @@ bool emberAfWindowCoveringClusterUpOrOpenCallback(app::CommandHandler * commandO
     {
         Attributes::TargetPositionTiltPercent100ths::Set(endpoint, WC_PERCENT100THS_MIN_OPEN);
     }
+
+    Delegate * delegate = GetDelegate(endpoint);
+    if (delegate)
+    {
+        if (HasFeature(endpoint, Feature::kPositionAwareLift))
+        {
+            LogErrorOnFailure(delegate->HandleLiftMovement());
```

Comment:
The ember callbacks seem to know if Up/Open or Down/Close is called, yet the handle callbacks only say "movement/tilt".

Why the split? Also why is there an ordering of features: if you only are position aware you get 'Lift' and otherwise if you have tilt you get 'tilt' callback. What if you have both features?

Have we considered passing in arguments into the delegate instead (like up/down constants)? Why would the delegate not have a single callback instead of separate lift/tilt?

Reply:
Yes, we can merge these two methods with passing argument.

According to spec, when we adjust the lift position, the tilt position should also be adjusted accordingly if Tilt feature is supported. If we have both features supported, we only need to adjust one type of position, so method HandleMovement should only be called only once.